tutorials/intermediate/imshow_extent.py

```diff
@@ -0,0 +1,179 @@
+"""
+*origin* and *extent* in `~.Axes.imshow`
+========================================
+
+:meth:`~.Axes.imshow` allows you to render an image (either a 2D array
+which will be color-mapped (based on *norm* and *cmap*) or and 3D RGB(A)
+array which will be used as-is) to a rectangular region in dataspace.
+The orientation of the image in the final rendering is controlled by
+the *origin* and *extent* kwargs (and attributes on the resulting
+`~.AxesImage` instance) and the data limits of the axes.
+
+The *extent* kwarg controls the bounding box in data coordinates that
+the image will fill specified as ``(left, right, bottom, top)`` in
+**data coordinates**, the *origin* kwarg controls how the image fills
+that bounding box, and the orientation in the final rendered image is
+also affected by the axes limits.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.gridspec import GridSpec
+
+
+def generate_imshow_demo_grid(auto_limits, extents):
```

Comment:
Might be worth having these as keyword arguments? When I read the tutorial I scrolled past this big block of code, and then was confused as to what `generate_imshow_demo_grid(True...)` did, but I think `generate_imshow_demo_grid(auto_limits=True...)` would be much better.